lib/android/app/src/main/java/com/reactnativenavigation/views/stack/topbar/titlebar/TitleSubTitleLayout.kt

```diff
@@ -15,10 +15,19 @@ import com.reactnativenavigation.options.Alignment
 import com.reactnativenavigation.options.FontOptions
 import com.reactnativenavigation.options.parsers.TypefaceLoader
 
+
 class TitleSubTitleLayout(context: Context) : LinearLayout(context) {
 
-    private val titleTextView = TextView(context).apply { TextViewCompat.setTextAppearance(this, R.style.TitleBarTitle) }
-    private val subTitleTextView = TextView(context).apply { TextViewCompat.setTextAppearance(this, R.style.TitleBarSubtitle) }
+    private val titleTextView = TextView(context).apply {
```

Comment:
Would be nice to declare the layout in XML and move all of these boilerplate to the XML. Then you can set the layout like so:

```kotlin
init {
    LayoutInflater.from(context).inflate(R.layout.title_layout, this)
}
```

Reply:
we adding one step of parsing the XML to the process, which does not have any added value, but, the style as I see don't have a special thing, I can do it in code.

Reply:
> which does not have any added value

I wasn't clear enough in my comment, but you should ask yourself if you're rushing into making bold statements. Before stating there's no value, ask what's the added value.

That being said, the idea is to move the boring styling logic out of the way. Handle initial style configurations in XML, keep logic in code.

Reply:
yeah, there is an added value sure, sorry for rushing out for my statement :].

My point is that moving them to XML will group them in one place and separate styling from the view logic, but here we add a step that is parsing XML, which I don't prefer while having a better solution, extract the style to another function out of this file, which its responsibility is to create the styles, wdyt?

Reply:
> moving them to XML will group them in one place and separate styling from the view logic

Yes, exactly. Move the boring layout/style/configuration code out of the way. Keep the interesting stuff in code. 

> we add a step that is parsing XML, which I don't prefer 

I don't understand - what's the downside of declaring a 20 LOC XML file and then inflating it? 🤔

Reply:
hmm, the downside is really not that big difference, but, that small difference is: instead of inflating -> setting attributes, using custom view styled for those attributes saves us from adding a Lil step (parsing XML) which is in this case, 3 attributes only, that makes XML parsing not that necessary, when actually we can set those attributes directly.

Besides, I tried the xml to set those attributes and it seems that `TextViewCompat.setTextAppearance(this, R.style.TitleBarTitle)` does not affect `singleLine` or `maxLines` attribute, see:

![Screen Shot 2021-04-27 at 6 13 30](https://user-images.githubusercontent.com/7227793/116180812-354e9080-a722-11eb-982e-646917a27f27.png)